### PR TITLE
feat(p-tag): renewal p-tag

### LINKED ISSUE #71: [Tag] Add category, key(label) spec
**Is your feature request related to a problem? Please describe.**
There is no way to represent data in tags in different ways.

**Describe the solution you'd like**
Add category, key(label) spec
image

[Figma](https://www.figma.com/file/IS6P8y1Wn2nfBC4jGlSiya/Components?node-id=10%3A47318)

### PR DESCRIPTION
**Pull request by piggggggggy**
### To Reviewers
- [ ] Skipping review is not a problem (```style```, ```chore``` ONLY)
- [ ] Not that difficult

### Type of Change
- [ ] New feature
- [ ] Bug fixes
- [x] Feature improvement
  - [x] These changes require a usage change
- [ ] Others (performance improvement, refactoring, CI/CD, etc.)

### Checklist
- [x] Updated Storybook documents
- [ ] Tested with console

### Description
- add 'Category' spec
- add `keyItem`, `valueItem` and `categoryItem` props.
- add `key`, `value` and `category` slots.
- update storybook
- add mock data

image

### CODE REVIEW

File: src/data-display/tags/PTag.vue [hunk 1]
```diff
@@ -73,16 +94,19 @@ export default defineComponent<Props>({
     display: inline-flex;
     align-items: center;
     overflow: hidden;
-    padding: 0.125rem 0.25rem 0.125rem 0.5rem;
+    padding: 0.0625rem 0.5rem 0.125rem;
```
Comment (minseolee) [inline src/data-display/tags/PTag.vue:99]: Is this value (`0.0625`) correct ? Would you like to check one more time please ?

File: src/data-display/tags/PTag.vue [hunk 2]
```diff
@@ -112,12 +136,12 @@ export default defineComponent<Props>({
             @apply bg-red-200 border-red-500;
         }
     }
-    .text {
-        font-size: 0.75rem;
-        line-height: 1.4;
-        white-space: normal;
-        word-break: break-word;
-        max-width: 100%;
```
Comment (minseolee) [inline src/data-display/tags/PTag.vue:120]: I think removing these css attributes can make some side effects.
Such as text overflow. Is it OK ?

**Review by minseolee — APPROVED** [review]
Great ! Left some comments

File: src/data-display/tags/story-helper.ts [hunk 3]
```diff
@@ -91,16 +136,52 @@ export const getTagArgTypes = (): ArgTypes => ({
...
             type: {
                 summary: null,
             },
-            defaultValue: {
+            category: 'slots',
+        },
+    },
+    categorySlot: {
+        name: 'category',
+        description: 'Slot for replace category item',
```
Comment (WANZARGEN) [inline src/data-display/tags/story-helper.ts]:
```suggestion
        description: 'Slot to replace category item',
```

I think it must be better :)

File: src/data-display/tags/story-helper.ts [hunk 4]
```diff
@@ -91,16 +136,52 @@ export const getTagArgTypes = (): ArgTypes => ({
             type: 'text',
         },
     },
-    //
-    default: {
```
Comment (WANZARGEN) [inline src/data-display/tags/story-helper.ts:95]: 👍

File: src/data-display/tags/story-helper.ts [hunk 5]
```diff
@@ -1,6 +1,51 @@
...
+        defaultValue: undefined,
+        table: {
+            type: {
+                summary: 'object',
+            },
+            category: 'props',
+            defaultValue: {
+                summary: undefined,
+            },
+        },
```
Comment (WANZARGEN) [inline src/data-display/tags/story-helper.ts:49]: I think that for the playground story, we need control for all props.

File: src/data-display/tags/PTag.vue [hunk 6]
```diff
@@ -9,7 +9,11 @@
              width="0.8rem"
              height="0.8rem"
         />
-        <span class="text"><slot /></span>
+        <slot name="default">
+            <span v-if="categoryItem" class="category"><slot name="category">[{{ categoryItem.name || categoryItem.label }}]</slot></span>
```
Comment (WANZARGEN) [inline src/data-display/tags/PTag.vue]:
```suggestion
            <span v-if="categoryItem" class="category"><slot name="category">[{{ categoryItem.label || categoryItem.name }}]</slot></span>
```

I think that `label` has higher priority than `name`.

**Review by WANZARGEN — COMMENTED** [review]
I left some comments! Plz check it and if you have other opinion, let's discuss about that!

File: src/data-display/tags/PTag.vue (hunk 2 above)
Comment (piggggggggy) [reply src/data-display/tags/PTag.vue:120]: Thanks, i'll handle that.

**Comment by piggggggggy** [comment]
@minseolee @WANZARGEN 
I reflected your opinions. 
- fix some awkward value in style
- prevent side effects that may happen
- fix awkward context in docs
- change priority between `name` and `label`
- other points to improve

Thank you for reviewing!

**Pull request merged**